code/iaas/logic/src/main/java/io/cattle/platform/process/instance/InstancePreStart.java

```diff
@@ -66,7 +66,12 @@ public void doWithLockNoResult() {
             Map<String, Object> driver_opts = DataAccessor.fieldMap(v, VolumeConstants.FIELD_VOLUME_DRIVER_OPTS);
             if (driver_opts.containsKey(VolumeConstants.EC2_AZ)) {
                 String zone = driver_opts.get(VolumeConstants.EC2_AZ).toString();
-                labels.put(HostAffinityConstraint.LABEL_HEADER_AFFINITY_HOST_LABEL, String.format("%s=%s", VolumeConstants.HOST_ZONE_LABEL_KEY, zone));
+                String label = String.format("%s=%s", VolumeConstants.HOST_ZONE_LABEL_KEY, zone);
+                if (labels.containsKey(HostAffinityConstraint.LABEL_HEADER_AFFINITY_HOST_LABEL)) {
```

Comment:
To protect against the (unlikely but still possible) scenario of `labels.get(HostAffinityConstraint.LABEL_HEADER_AFFINITY_HOST_LABEL)` returning NULL and thus getting an NPE on the next line, could you change this if statement to !StringUtils.isEmpty(labels.get(HostAffinityConstraint.LABEL_HEADER_AFFINITY_HOST_LABEL)))`?